addons/web/static/tests/fields/basic_fields_tests.js

```diff
@@ -479,6 +479,87 @@ QUnit.module('basic_fields', {
         form.destroy();
     });
 
+    QUnit.module('FieldNumeric');
+
+    QUnit.test('numeric field: fields with keydown on numpad decimal key', async function (assert) {
+        assert.expect(6);
+
+        this.data.partner.fields.float_factor_field = { string: "Float Factor", type: 'float_factor' };
+        this.data.partner.records[0].float_factor_field = 9.99;
+
+        this.data.partner.fields.float_time_field = { string: "Float Time", type: 'float_time' };
+        this.data.partner.records[0].float_time_field = 9.99;
+
+        this.data.partner.fields.monetary = { string: "Monetary", type: 'monetary' };
+        this.data.partner.records[0].monetary = 9.99;
+        this.data.partner.records[0].currency_id = 1;
+
+        this.data.partner.fields.percentage = { string: "Percentage", type: 'percentage' };
+        this.data.partner.records[0].percentage = .99;
+
+        const form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: `
+                <form string="Partners">
+                    <field name="float_factor_field" options="{'factor': 0.5}"/>
+                    <field name="qux"/>
+                    <field name="float_time_field"/>
+                    <field name="int_field"/>
+                    <field name="monetary"/>
+                    <field name="currency_id" invisible="1"/>
+                    <field name="percentage"/>
+                </form>
+            `,
+            res_id: 1,
+            translateParameters: {
+                decimal_point: "🇧🇪",
+            },
+        });
+
+        // Prepare the numpad "dot" keydown event
+        const ev = new KeyboardEvent('keydown', { code: 'NumpadDecimal' });
+        ev.getModifierState = (mod) => mod === 'NumLock';
```

Comment:
As I said earlier, this is annoying, but it may be the right thing to do though. (I don't know at this point)
why aren't key and code enough ? is it because key would output another symbol in another non-latin locale ?

Reply:
Depending on the system region settings, it may.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values#Numeric_keypad_keys